content/docs/pipelines/sdk/python-based-visualizations.md

```diff
@@ -0,0 +1,177 @@
+Kubeflow Pipelines Team and serve as a way for users to easily and quickly
+generate powerful visualizations. However, these are not limited to just the
+Pipelines Team and this will be discussed further below in
+[Why Predefined Visualizations are Important](/docs/pipelines/sdk/
+python-based-visualizations/#why-predefined-visualizations-are-important).
+The second category is **arbitrary visualizations**. Arbitrary visualizations
+allow for user provided python visualization code to be used to generate
+visualizations. These visualizations allow for rapid development,
+experimentation, and customizability when visualizing results.
+
+## Why Predefined Visualizations are Important
+
+Predefined visualizations provide major benefits for the Pipelines Team and
+users of Pipelines. The Pipelines Team is able to develop predefined
+visualizations to provide users of Pipelines with powerful, extensible, and
+standardized visualizations that work out of the box. For users of Pipelines,
+they are able to utilize existing predefined visualization and build their own.
+For some users, such as those who use Pipelines with a team or those who
+frequently generate complex and specialized visualizations, predefined
+visualizations can be leveraged to reduce development time and provide
+standardizations of visualized results. Details about how to create predefined
+visualizations can be found [below](/docs/pipelines/sdk/
+python-based-visualizations/#how-to-create-predefined-visualizations).
+
+## How to Create Predefined Visualizations
+
+1. Determine what the new visualization will be.
+2. Add a new type for the visualization within the.
+[visualization.proto](https://github.com/kubeflow/pipelines/blob
+/master/backend/api/visualization.proto#L78) file.
+    * The name of the visualization should be in screaming snake case
+    (i.e. `VISUALIZATION_NAME`).
+3. Run [`./generate_api.sh`](https://github.com/kubeflow/pipelines/blob/master
+/backend/api/generate_api.sh) within the `backend/api` directory.
+4. Download the [Swagger Codegen](https://swagger.io/tools/swagger-codegen/)
+jar file.
+    * Currently, version 2.3.1 is used to generate the api. Should this become
+    out of date, the version can be checked [here](https://github.com/kubeflow/
+    pipelines/blob/master/frontend/src/apis/visualization/.swagger-codegen/
+    VERSION).
+    * This step is only required if the Swagger Codegen jar file is not present
+    in the `frontend` directory. If you already have the jar file, you can skip
+    steps 5 and 6.
+5. Place the Swagger Codegen jar file in the `frontend` directory.
+6. Rename the Swagger Codegen har file to **swagger-codegen-cli.jar**.
+7. Run `npm run apis:visualization` within the `frontend` directory.
+8. Create a new python file that will be executed to generate a visualization.
+    * Python 3 **MUST** be used.
+    * The new python file should be created within the
```

Comment:
Change passive voice to active voice here and throughout this PR. See the style guide for more info. Passive voice is by its nature ambiguous. For example, in this case do we mean that the user must create the file in that directory or some routine should do that, and do we mean that the user must use dependency injection or that KFP does that?